website/components/FoodCategory/index.tsx

```diff
@@ -15,10 +15,10 @@ import "swiper/css";
 import "swiper/css/navigation";
 import CategorySwiperSkeleton from "../SkeletonPlaceholders/CategorySwiper";
 
-// Manage twice data calling issue using this flag
-let dataCalled: any = false;
-
 const FoodCategory = () => {
+  // Manage twice data calling issue using this flag
+  let dataCalled: any = false;
```

Comment:
Should it be manage based on the retrieved data is empty or not? It's not the best practice to manage based on flag.

Reply:
yaa we can do it but the state management is taking time to set data and meanwhile it's calling twice, that why added flag to manage it smoothly.
inshort it's calling twice in micro second that's why I need to manage it using hard thing

Reply:
yaa

Reply:
then it should be called once only i guess.

Reply:
hmm, but it's calling twice

Reply:
I think we should investigate what is causing recalling, instead of preventing it with flag.